src/foam/graphics/TreeNode.js

```diff
@@ -0,0 +1,297 @@
+            right: o.right + x,
+          }));
+
+          for (var j = 0; j < childOutline.length; j++) {
+            champion[j] = champion[j] || {
+              left: Number.MAX_SAFE_INTEGER,
+              right: Number.MIN_SAFE_INTEGER
+            };
+            champion[j].left = Math.min(childOutline[j].left, champion[j].left);
+            champion[j].right = Math.max(childOutline[j].right, champion[j].right);
+          }
+        }
+
+        var totalOutline = rootLevelOutline.concat(champion);
+
+        return totalOutline;
+      }
+    },
+    { name: 'height', factory: function () { return this.nodeHeight; } },
+    { name: 'width', factory: function () { return this.nodeWidth; } },
+    ['border', 'gray'],
+    {
+      name: 'childNodes',
+      factory: function () { return []; }
+    },
+    ['left', 0],
+    ['right', 0],
+    ['maxLeft', 0],
+    ['maxRight', 0],
+    {
+      name: 'x',
+      expression: function(neighborX, centerX) {
+        return neighborX + centerX;
+      },
+      postSet: function() {
+        this.neighborX = 0;
+        this.centerX = 0;
+        this.x = undefined;
+      }
+    },
+    {
+      class: 'Float',
+      name: 'neighborX'
+    },
+    {
+      class: 'Float',
+      name: 'centerX'
+    },
+    {
+      class: 'Boolean',
+      name: 'expanded',
+      expression: function(childNodes) {
+        return childNodes.length < 5;
+      },
+      postSet: function() {
+        this.graph.doLayout();
+      }
+    },
+    ['color', 'white']
+  ],
+
+  methods: [
+    function initCView() {
+      this.SUPER();
+
+      this.formatNode();
+
+      if (this.relationship) {
+        var data = this.data.clone(this.__subContext__);
+        try {
+          data[this.relationship.forwardName].select(data => {
+            this.addChildNode({ data: data });
+          });
+        } catch (x) { }
+        this.graph.doLayout();
+      }
+
+      this.canvas.on('click', (e) => {
+        var p = this.parent;
+
+        while ( this.cls_.isInstance(p) ) {
+          if ( ! p.expanded ) return;
+          p = p.parent;
+        }
+
+        var point = this.globalToLocalCoordinates({
+          w: 1,
+          x: e.offsetX,
+          y: e.offsetY,
+        });
+        point.x += this.width / 2;
+
+        var hit = this.hitTest(point)
+        if ( hit ) {
+          this.expanded = ! this.expanded;
+        }
+      });
+    },
+
+    function paint(x) {
+      if (!this.parentNode || this.parentNode.expanded) this.SUPER(x);
+    },
+
+    function paintSelf(x) {
+      x.save();
+
+      x.translate(-this.width / 2, 0);
+      this.SUPER(x);
+
+      // reset translate and shadow settings
+      x.restore();
+
+      this.paintConnectors(x);
+    },
+
+    function paintConnectors(x) {
+      function line(x1, y1, x2, y2) {
+        x.beginPath();
+        x.moveTo(x1, y1);
+        x.lineTo(x2, y2);
+        x.stroke();
+      }
+
+      x.lineWidth = 0.5; //this.borderWidth;
```

Comment:
What's with the commented out code?

Reply:
I think that comment is there to say that the lineWidth is equal to borderWidth

Reply:
K im just gunna create a property called lineWidth instead borderWidth doesn't even exist in treeNode

Reply:
Yeah so when the lines are equal to borderWidth the line shows up a bit thicker, i'll expose a property called lineWidth to TreeNode and set that to 0.5. I noticed afterwards lineWidth gets reset to borderWidth in order to draw the collapse/expand arrows, ill leave that to be set to borderWidth.